pkg/cortex/serve/cortex_internal/lib/api/batch.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2021 Cortex Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from copy import deepcopy
+import imp
+import inspect
+import os
+import dill
+from typing import Any, Dict, Optional
+
+from datadog.dogstatsd.base import DogStatsd
+
+from cortex_internal.lib import util
+from cortex_internal.lib.client.tensorflow import TensorFlowClient
+from cortex_internal.lib.model import TFSAPIServingThreadUpdater
+from cortex_internal.lib.api.utils import CortexMetrics
+from cortex_internal.lib.api.validations import (
+    are_models_specified,
+    validate_class_impl,
+)
+from cortex_internal.lib.type import (
+    PythonHandlerType,
+    TensorFlowNeuronHandlerType,
+    TensorFlowHandlerType,
+    handler_type_from_api_spec,
+)
+from cortex_internal.lib.exceptions import CortexException, UserException, UserRuntimeException
+
+PYTHON_CLASS_VALIDATION = {
+    "required": [
+        {
+            "name": "__init__",
+            "required_args": ["self", "config"],
+            "optional_args": ["job_spec", "metrics_client"],
+        },
+        {
+            "name": "handle_batch",
+            "required_args": ["self"],
+            "optional_args": ["payload", "batch_id"],
+        },
+    ],
+    "optional": [
+        {"name": "handle_on_job_complete", "required_args": ["self"]},
```

Comment:
`handle_on_job_complete` sounds too cumbersome, what do you think of keeping the old name?